sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/json/JsonScanBuilder.scala

```diff
@@ -29,8 +32,26 @@ class JsonScanBuilder (
     schema: StructType,
     dataSchema: StructType,
     options: CaseInsensitiveStringMap)
-  extends FileScanBuilder(sparkSession, fileIndex, dataSchema) {
+  extends FileScanBuilder(sparkSession, fileIndex, dataSchema) with SupportsPushDownFilters {
   override def build(): Scan = {
-    JsonScan(sparkSession, fileIndex, dataSchema, readDataSchema(), readPartitionSchema(), options)
+    JsonScan(
+      sparkSession,
+      fileIndex,
+      dataSchema,
+      readDataSchema(),
+      readPartitionSchema(),
+      options,
+      pushedFilters())
   }
+
+  private var _pushedFilters: Array[Filter] = Array.empty
+
+  override def pushFilters(filters: Array[Filter]): Array[Filter] = {
+    if (sparkSession.sessionState.conf.csvFilterPushDown) {
```

Comment:
Ur, this should be `jsonFilterPushDown`. It seems that we only test the positive case because both csv/json conf is true by default. Please fix this and add a test case.

Reply:
I am even not sure that this check is need because:
1. Actual disabling/enabling happens here:  https://github.com/apache/spark/pull/27366/files#diff-467cfe570f88d176aed53f30c3766356R67
2. The upper layer applies filters again.

@dongjoon-hyun In any case, thank you for detecting this, and start reviewing the PR.

Reply:
fixed and added a test to JsonV2Suite

Reply:
Thanks.